sdk/src/main/scala/coop/rchain/sdk/dag/merging/data/MergeableScope.scala

```diff
@@ -0,0 +1,60 @@
+package coop.rchain.sdk.dag.merging.data
+import cats.syntax.all._
+import coop.rchain.sdk.dag.merging.data.MergeableScope.FinalizedDeploys
+
+final case class MergeableScope[B, D](
+    latestMsgs: Set[B],
+    finalityFringe: Set[B],
+    finalityState: Map[B, FinalizedDeploys[D]]
+)(implicit bAsBlock: B => Block[B, D]) {
+  // to not do .map(msgSet) all the time
+  implicit val bSetAsBlock: Set[B] => Set[Block[B, D]] = (bs: Set[B]) => bs.map(bAsBlock)
```

Comment:
Please use `def` to define functions.

---

This is also very difficult to read and IMO unnecessary. If there is a mapping it should be written.